pg_search/src/postgres/customscan/pdbscan/projections/snippet.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2023-2024 Retake, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use crate::api::index::FieldName;
+use crate::api::search::{DEFAULT_SNIPPET_POSTFIX, DEFAULT_SNIPPET_PREFIX};
+use crate::index::state::SearchState;
+use crate::nodecast;
+use crate::postgres::customscan::pdbscan::projections::OpaqueRecordArg;
+use pgrx::pg_sys::expression_tree_walker;
+use pgrx::{
+    default, direct_function_call, pg_extern, pg_guard, pg_sys, FromDatum, IntoDatum, PgList,
+};
+use std::ptr::addr_of_mut;
+use tantivy::snippet::SnippetGenerator;
+use tantivy::DocAddress;
+
+#[derive(Debug, Eq, Hash, PartialEq)]
+pub struct SnippetInfo {
+    pub field: FieldName,
+    pub start_tag: String,
+    pub end_tag: String,
+    pub max_num_chars: usize,
+}
+
+#[pg_extern(name = "snippet", stable, parallel_safe)]
+fn snippet_from_relation(
+    _relation_reference: OpaqueRecordArg,
+    field: FieldName,
+    start_tag: default!(String, "'<b>'"),
+    end_tag: default!(String, "'</b>'"),
```

Comment:
@rebasedming this is the `paradedb.snippet()` function.  It'll need documenting.